docs/API/fliplet-ui-actions.md

```diff
@@ -0,0 +1,71 @@
+# `Fliplet.UI.Actions()`
+
+(Returns **`Promise`**)
+
+Show a sheet of options the user can choose from.
+
+* iOS uses the native `UIActionSheet`
+* Android uses the native `AlertDialog`
+* WP8 uses the native `Popup`
+* Web uses a custom dialog (TBC)
+
+If an action is chosen, the Promise is resolved when the specified action is completed or resolved with the 1-based index provided as the first parameter. The Promise is also resolved when user chooses to cancel and dismiss the action sheet. In this case, the index provided in the resolving function will be `undefined`.
+
+## Screenshots
+
+(Coming soon...)
+
+## Usage
+
+```js
+Fliplet.UI.Actions(options)
```

Comment:
`Actions.show()` or `.actions()` might be better suited. I'd recommend the first so you can add more methods in the future. Either ways, my preference for "actions.show" is also because "actions" is neither a verb nor an action.

Another way you can view this:

```js
var actions = Fliplet.UI.Actions(...);

actions.show();

actions.then(function (result) {
  // on complete
});

// if you need to
actions.dismiss();

acitions.onDismiss(fn); // or reject the promise

// also chained since it can return the same promise
actions.show().then(fn);
```

Reply:
`Fliplet.UI.Toast()` opens the Toast UI right away, while the returning instance (in the promise resolving function) contains a `.dismiss()` method for users to programmatically dismiss the UI.

You are suggesting a different pattern here from `Fliplet.UI.Toast()`. This either:

1. Creates more work for me because I'd need to update `Fliplet.UI.Toast()`, its documentation and its implementation in `fliplet-core/navigate.js` 😞; or
2. Creates an inconsistency in how `Fliplet.UI` is used

What are your thoughts or preference?

Reply:
I don't mind doing more work as long we get a much longer term perspective rather than compounding the work required every time we add a new function to this :-/

Reply:
BTW this would be my preference based on your various suggestions.

I don't see how `.dismiss()` would be useful at all. `alert()` doesn't have a dismiss. This is kind of the same.

```js
var actions = Fliplet.UI.Actions(options);
actions.show().then(function onActionComplete (i) {
  // i will be an 1-based index or undefined if Cancel is selected
});
```

Reply:
@tonytlwu OK I understand the concern on consistencies between other parts of `Fliplet.UI`. My main doubt is around not using actions to describe an action. 

Coming with a software background, I expect `Toast()` to simply create a new instance of such class because:
- it starts with an uppercase letter
- it's a name, not an action/verb

Hence why the instance exposes a `show` property in my example. It's also similar to:

```
dataSources.connect() -> connection.insert()
```

If we talk about consistency, `Fliplet.UI` is inconsistent with any other JS API.

Reply:
@tonytlwu I like the one you just posted, plus you can daisy chain quite easily:

```js
Fliplet.UI.Actions(options).show().then(function onActionComplete (i) {

});
```

Reply:
@tonytlwu I know it feels nonsense to having to explicitly call `show` if you only have one action and it's slower to write. I'll let you brainstorm and decide whether to stick with `Actions().then` or `Actions().show().then`.

A solution to this is that `Actions()` shows by default unless `Actions({ show: false })` (if you ever need to)

Reply:
@squallstar Gotcha. I do wish you had caught me out earlier on how `Fliplet.UI` functions has been different from the other JS APIs.

Does this mean I should 1️⃣  create a `Flipet.UI.Toast.create()` or `Flipet.UI.Toast.new()` function to create an instance in order to match consistency with all other JS APIs? Or can we 2️⃣  allow this to be different because all other JS APIs are non-UI related, which means they provided a namespace to interact with a model, e.g. `Fliplet.DataSource.connect()` connects with a data source.

My preference is 2️⃣ and I'm happy to go with:

```js
Fliplet.UI.Actions(options).then(function onActionComplete (i) {

});
```

...because I can't ever imagine you'd want a list of actions and not show it. It's like creating an `alert|()` or `prompt()` but not show it. It's a relatively simple choice of actions for the user to make.